sfge/src/test/java/com/salesforce/graph/ops/expander/ApexPathExpanderTest.java

```diff
@@ -151,6 +151,65 @@ public void testRegistryCleanup() {
         MatcherAssert.assertThat(path.getInvocableVertexToPaths().entrySet(), hasSize(equalTo(3)));
     }
 
+    @Test
+    public void testLazyExpansion() {
+        // Set the maximum expander registry size to a very low value.
+        setPathRegistryLimit(5);
+        // spotless:off
+        String sourceCode =
+            "public class MyClass {\n"
+                // This method has only one fork, but that fork has 8 branches.
```

Comment:
For my understanding, can you tell me how there are 8 branches? Since there are three variables each with an if/else, I was expecting 6 branches.

Reply:
@jag-j , the method forks into two paths at the first `if`, then four at the second, and eight at the third.

Reply:
Never mind, I get it now. It is 2^3 forks we are talking about.